app/models/post_mover.rb

```diff
@@ -342,9 +341,12 @@ def copy_topic_users
                               notifications_changed_at, notifications_reason_id)
       SELECT tu.user_id,
              :new_topic_id                               AS topic_id,
-             CASE
-               WHEN p.user_id IS NULL THEN FALSE
-               ELSE TRUE END                             AS posted,
+               EXISTS(
+                 SELECT 1
+                 FROM posts p
+                 WHERE p.topic_id = :new_topic_id
+                   AND p.user_id = tu.user_id
```

Comment:
I think it auto optimises, but you might as well do a `limit 1`

once done feel free to merge

Reply:
yes, `EXISTS` aborts the subquery as soon as one row is found

> The subquery will generally only be executed long enough to determine whether at least one row is returned